### PR TITLE
Using node resolver to locate script dependencies

### PR DESCRIPTION
**Pull request by morganick**
When you're working with Monorepos, the scripts you are looking for may be hoisted into a parent directory. These updates take advantage of Node's resolver to locate the package and base our lookup for the scripts from that location.

## To-Do:
- [x] Chat with @jamonholmgren about whether this is the main option or a side note in the README

### CODE REVIEW

**Review by jamonholmgren — APPROVED** [review]

**Review by jamonholmgren — CHANGES_REQUESTED** [review]
We need to update `scripts/install-colo-loco` too, so it makes these changes.

I'll do that and push up an update to this branch.

**Comment by jamonholmgren** [comment]
Update here, please review! https://github.com/jamonholmgren/react-native-colo-loco/pull/30/commits/2443531a29f7ef498c331ce73bff6e5135f413ed

**Comment by jamonholmgren** [comment]
@morganick I'm not sure why my newest commit didn't make it into this PR. https://github.com/morganick/react-native-colo-loco/commit/81c38b239ede18ee3f872d82d8535b05ebc1a9b9

**Comment by jamonholmgren** [comment]
Now they're there!

**Review by morganick — COMMENTED** [review]
There's a typo in the pathing for the iOS scripts. Will push a commit up shortly to fix.

File: README.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ _NOTE: If this doesn't work or you have a non-standard project structure, try th
 For iOS, add this to your Podfile (`ios/Podfile`) (don't forget to change `MyApp` to your actual app name):
 
 ```ruby
-require_relative '../node_modules/react-native-colo-loco/scripts/ios.rb'
+require File.join(File.dirname(`node --print "require.resolve('react-native-colo-loco/package.json')"`), "scripts/ios.rb")
```
[the remark this answers is not in the document]
Comment (morganick) [reply README.md]: 🤦 forgot this was using `dirname` from ruby... ignore. More ☕ required.

File: test/install-colo-loco.test.js [hunk 2]
```diff
@@ -34,13 +34,15 @@ describe("Checking for install-colo-loco. 🤪", () => {
     // iOS set up is correct
     // Podfile
     const podfile = await fs.readFile(`${appPath}/ios/Podfile`, "utf8")
-    expect(podfile).toContain("react-native-colo-loco/scripts/ios.rb")
+    expect(podfile).toContain("react-native-colo-loco/package.json")
+    expect(podfile).toContain("scripts/ios.rb")
```
[the remark this answers is not in the document]
Comment (morganick) [reply test/install-colo-loco.test.js:38]: ignore...

File: test/install-colo-loco.test.js [hunk 3]
```diff
@@ -34,13 +34,15 @@ describe("Checking for install-colo-loco. 🤪", () => {
...
     expect(podfile).toContain("link_colocated_native_files")
 
     // Android set up is correct
     // settings.gradle
     const settingsGradle = await fs.readFile(`${appPath}/android/settings.gradle`, "utf8")
-    expect(settingsGradle).toContain("react-native-colo-loco/scripts/android.groovy")
+    expect(settingsGradle).toContain("react-native-colo-loco/package.json")
+    expect(settingsGradle).toContain("scripts/android.groovy")
```
[the remark this answers is not in the document]
Comment (morganick) [reply test/install-colo-loco.test.js:45]: I think it be more consistent to leave as is. The `../` is an implementation detail.

File: scripts/install-colo-loco [hunk 4]
```diff
@@ -116,7 +116,7 @@ function setupIOS(appName, sourceFolder, _packageName) {
     const coloLocoPodfileContents = `
 # React Native Colo Loco autolinks native files colocated next to your JS files
 # More info here: https://github.com/jamonholmgren/react-native-colo-loco
-require_relative '../node_modules/react-native-colo-loco/scripts/ios.rb'
+require File.join(File.dirname(\`node --print "require.resolve('react-native-colo-loco/package.json')"\`), "scripts/ios.rb")
```
[the remark this answers is not in the document]
Comment (morganick) [reply scripts/install-colo-loco:119]: Ignore... I was thinking of the android side. We're using `dirname` here so we're g2g.

**Comment by morganick** [comment]
Tests pass and I've reviewed the rest of the changes 👍

File: test/install-colo-loco.test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jamonholmgren) [reply test/install-colo-loco.test.js:45]: Good call.

**Review by jamonholmgren — APPROVED** [review]

**Pull request merged**